### PR TITLE
cat info command

### PR DESCRIPTION
**Pull request by VishramKidPG123**
lore from thetrashcell!!
(he might rewrite it specifically for this idk)

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -1082,6 +1085,25 @@ async def info(message: discord.Interaction):
...
+@discord.app_commands.describe(cat_type="Select a cat type")
+@discord.app_commands.autocomplete(cat_type=cat_type_autocomplete)
+async def catinfo(message: discord.Interaction, cat_type: str):
+    if cat_type not in cattypes:
+        await message.response.send_message("cat does not exist", ephemeral=True)
+        return
+    icon = get_emoji(cat_type.lower() + "cat")
+    s = sum(type_dict.values())
+    v = type_dict[cat_type]
+    embed = discord.Embed(title=f"{icon} {cat_type} Cat Info", color=0x6E593C) # possibly make color different per cat type
```
Comment (milena-kos) [inline main.py:1099]: maybe remove "Info"? will look smoother imo

File: main.py [hunk 2]
```diff
@@ -1082,6 +1085,25 @@ async def info(message: discord.Interaction):
...
+    embed.add_field(name="LORE", value=lore[cat_type], inline=False)
```
Comment (milena-kos) [inline main.py:1100]: yelling not nessecary :staring_cat:

File: main.py [hunk 3]
```diff
@@ -1082,6 +1085,25 @@ async def info(message: discord.Interaction):
...
+    embed.add_field(name="You have:", value=get_cat(message.guild.id, message.user.id, cat_type), inline=True)
+    embed.add_field(name="Spawn Chance", value=f"{round(v / s * 100, 2)}%", inline=True)
+    embed.add_field(name="Trade Value", value=str(round(s / v, 1)), inline=True)
+    embed.set_footer(text=f"lore from the one and only, {gen_credits['trash']}!!")
```
Comment (milena-kos) [inline main.py:1104]: looks a bit redundant, maybe remove this and add lore credit to /info

File: main.py [hunk 4]
```diff
@@ -1082,6 +1085,25 @@ async def info(message: discord.Interaction):
...
+async def catinfo(message: discord.Interaction, cat_type: str):
+    if cat_type not in cattypes:
+        await message.response.send_message("cat does not exist", ephemeral=True)
+        return
+    icon = get_emoji(cat_type.lower() + "cat")
+    s = sum(type_dict.values())
+    v = type_dict[cat_type]
+    embed = discord.Embed(title=f"{icon} {cat_type} Cat Info", color=0x6E593C) # possibly make color different per cat type
+    embed.add_field(name="LORE", value=lore[cat_type], inline=False)
+    embed.add_field(name="You have:", value=get_cat(message.guild.id, message.user.id, cat_type), inline=True)
```
Comment (milena-kos) [inline main.py:1101]:
```suggestion
    embed.add_field(name="You have", value=get_cat(message.guild.id, message.user.id, cat_type), inline=True)
```

File: main.py [hunk 5]
```diff
@@ -1082,6 +1085,25 @@ async def info(message: discord.Interaction):
         embedVar.set_footer(text="Last code update:")
     await message.followup.send(embed=embedVar)
 
+@bot.tree.command(description="View information about a cat")
+@discord.app_commands.rename(cat_type="type")
+@discord.app_commands.describe(cat_type="Select a cat type")
+@discord.app_commands.autocomplete(cat_type=cat_type_autocomplete)
+async def catinfo(message: discord.Interaction, cat_type: str):
```
Comment (milena-kos) [inline main.py:1092]: maybe a better name? this looks a bit ugly

File: main.py (hunk 2 above)
Comment (VishramKidPG123) [reply main.py:1100]: matpat reference

Comment (frinkifail) [reply main.py:1100]: LOREEEEEEEEEEEEEEEEEE

File: main.py (hunk 5 above)
Comment (VishramKidPG123) [reply main.py:1092]: yeah I thought about the name for a bit; idk what to make it

**Comment by aar0n-py** [comment]
holy moly lots of code idk what this is again i really only get async def literally

**Comment by milena-kos** [comment]
buddy this is barely any code its mostly hardcoded lore values and 5 lines to make an embed out of them :staring_cat:

**Comment by VishramKidPG123** [comment]
> buddy this is barely any code
so true

**Comment by milena-kos** [comment]
im currently waiting on whether trash will rewrite this or not

**Comment by NathanCrazy** [comment]
infinite years later :shock:

**Comment by RiskLM** [comment]
I literally made a account just to reply to this with this

**Comment by RiskLM** [comment]
Day 205 of asking for 6 ultimates

**Comment by NathanCrazy** [comment]
oh wow that's a big diff :staring_cat:

**Comment by milenakos** [comment]
will also need to move lore file to config folder at some point

**Pull request closed without merging**